app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,64 @@
+class SalesController < ApplicationController
+  def index
+    @vendor = Vendor.find(params[:vendor_id])
+    @sales = @vendor.recent_sales_first
+
+    if @sales.length == 0
+      render :empty
+    end
+  end
+
+  def current_month
+    @vendor = Vendor.find(params[:vendor_id])
+    @sales = @vendor.month_recent_sales_first
+
+    @header = "#{Date::MONTHNAMES[Date.today.month]}'s Sales"
+    @link_name = "View All Sales"
+    @link = "/vendors/#{ @vendor.id }/sales"
+
+    if @sales.length == 0
+      render :empty
+    else
+      render :index
```

Comment:
Nice! You might consider making index a partial that `#index` and `#current_month` can load, to keep them a little less tightly coupled.